pkg/controller.v1beta1/suggestion/suggestionclient/suggestionclient.go

```diff
@@ -343,6 +343,9 @@ func (g *General) ConvertTrials(ts []trialsv1beta1.Trial) []*suggestionapi.Trial
 		if t.IsMetricsUnavailable() {
 			continue
 		}
+		if !t.IsObservationAvailable() && t.IsEarlyStopped() {
+			continue
+		}
```

Comment:
@shaowei-su Can we add a test for this condition to the [`TestSyncAssignments`](https://github.com/kubeflow/katib/blob/8190f9c69c6ae2763257e57a67206b725a363e1f/pkg/controller.v1beta1/suggestion/suggestionclient/suggestionclient_test.go#L76) func or to a new test function?

Reply:
Thanks @tenzen-y ! I added few more trails in the mock so this branching logic is also validated.

Btw, could help restart the failed unit tests? Looks like a flaky test failed.